pint/models/dispersion.py

```diff
@@ -34,7 +71,27 @@ def dispersion_delay(self, toas):
         except AttributeError:
             warn("Using topocentric frequency for dedispersion!")
             bfreq = toas['freq']
-        return self.DM.value * DMconst / bfreq**2
+
+        # Constant dm delay
+        dmdelay = self.DM.value * DMconst / bfreq**2
+
+        # Get DMX delays
+        epoch_ind = 1
+        while epoch_ind in self.DMX_mapping:
```

Comment:
Another way of doing this without looping over them every time that you call disperson_delay() would be to explicitly put TOAs into DMX ranges and then add a flag with the dmxrange number to each TOA.  You could do that once.  Then after that, you could use the astropy table groupby code to apply the appropriate DM delays.  I think that would save a whole bunch of array comparison operations during a fit.